sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1957,6 +1973,8 @@ pi_result piMemBufferCreate(pi_context Context, pi_mem_flags Flags, size_t Size,
                                               HostPtr, Size, nullptr, 0,
                                               nullptr));
       }
+    } else if ((Flags & PI_MEM_FLAGS_HOST_PTR_ALLOC) != 0) {
```

Comment:
You are not setting the `HostPtr` to the allocated memory, so you can't get here, right? (line 1962 should be false). So why is this change needed?

Reply:
Maybe instead verify that `HostPtr` is null when PI_MEM_FLAGS_HOST_PTR_ALLOC is used?

Reply:
@smaslov-intel are you saying that hostPtr and PI_MEM_FLAGS_HOST_PTR_ALLOC is mutually exclusive? My understanding was that they are not.

Reply:
Yes, see in https://www.khronos.org/registry/OpenCL/sdk/2.2/docs/man/html/clCreateBuffer.html:

> CL_​MEM_​ALLOC_​HOST_​PTR and CL_​MEM_​USE_​HOST_​PTR are mutually exclusive.

PI_MEM_FLAGS_HOST_PTR_ALLOC is analogous to CL_​MEM_​ALLOC_​HOST_​PTR